apiserver/facades/client/charms/client.go

```diff
@@ -243,12 +216,15 @@ func (a *API) addCharmWithAuthorization(ctx context.Context, args params.AddChar
 	if err != nil {
 		return params.CharmOriginResult{}, errors.Trace(err)
 	}
+
+	a.logger.Debugf("AddCharm result: %+v", origin)
+
 	return params.CharmOriginResult{
 		Origin: origin,
 	}, nil
 }
 
-func (a *API) queueAsyncCharmDownload(ctx context.Context, args params.AddCharmWithAuth) (corecharm.Origin, error) {
```

Comment:
This method didn't queue the async charm download, it just adds a charm. To queue a charm download, you need a non-uploaded charm associated with an application.